### DISCUSSION TITLE
Status unstable

### DISCUSSION THREAD

**Discussion by UserPrivate** [serhmarch/ModbusTools#7] [Q&A]
Hi, im having hard time figuring out why when i read my reg which is on a esp32 i get mixed status. first scan its "Good" and the next is "BadNotCorrectRespons". Any input is appreciated!
![Capgggture](https://github.com/user-attachments/assets/845a6f14-9ba4-47f1-a6fa-7d4c497b33b9)

**Comment by serhmarch** [comment]
Hello. Try new 0.3.0 version. I have already encountered the “BadNotCorrectResponse” misuse before and fixed it

**Reply by UserPrivate** [reply]
Did try the new version but the issue persist..

**Comment by serhmarch** [comment]
1. First, check the client version (menu “Help->About”). Both “Version” and “ModbusLib version” should be “0.3.0”.

2.  Previously, this problem occurred if the client sent a `Tx` packet and waited for the `Rx` response and at this moment the server broke the current connection. 

Please note that when you receive a response (`Rx`-packet, e.g. 59.330), only 2 milliseconds pass and you make a new request (`Tx`-packet, 59.332). Perhaps your ESP32 cannot cope with such frequent requests and breaks the connection...
